packages/metadata/src/providers/localFileMetadata.provider.ts

```diff
@@ -0,0 +1,95 @@
+import { existsSync, readFileSync } from "fs";
+import { z } from "zod";
+
+import {
+    Cache,
+    ILogger,
+    Token,
+    TokenType,
+    ZKChainMetadata,
+    ZKChainMetadataItem,
+} from "@zkchainhub/shared";
+
+import { FileNotFound, IMetadataProvider, InvalidSchema } from "../internal.js";
+import { ChainSchema, TokenSchema } from "../schemas/index.js";
+
+export const LOCALFILE_METADATA_PREFIX = "local-metadata";
+
+/**
+ * Represents a local file metadata provider.
+ */
+export class LocalFileMetadataProvider implements IMetadataProvider {
+    /**
+     * Constructs a new instance of the LocalFileMetadataProvider class.
+     * @param tokenJsonPath The path to the token JSON file.
+     * @param chainJsonPath The path to the chain JSON file.
+     * @param logger The logger instance.
+     * @param cache The cache instance.
+     * @throws {FileNotFound} if any of the files is not found.
+     */
+    constructor(
+        private readonly tokenJsonPath: string,
+        private readonly chainJsonPath: string,
+        private readonly logger: ILogger,
+        private readonly cache: Cache,
+    ) {
+        if (!existsSync(tokenJsonPath)) {
+            throw new FileNotFound(tokenJsonPath);
+        }
+
+        if (!existsSync(chainJsonPath)) {
+            throw new FileNotFound(chainJsonPath);
+        }
+    }
+
```

Comment:
Lets add natspec on `IMetadataProvider` and reuse it on the classes that implement the interface